### PR TITLE
fix(core-select): missing clue filter in users directive under some conditions

### PR DESCRIPTION
**Pull request by GuillaumeNury**
## Description

- Extracted `CoreSelectKeyManager` to avoid duplication is simple/multi select panels. It makes Highlight logic more testable
- Fix issue in homonyms cache
- Fix highlight issue when using `LuCoreSelectUsersDirective`
- Fix missing clue when typing the first search when using `LuCoreSelectUsersDirective`

-----

Optionally, technical or more in-depth description for reviewers.
Keep an empty line under your text, as well as the 5 lines that follow it.

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2964/storybook

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2964/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/19812/ui-diff)

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/19813/ui-diff)

File: packages/ng/core-select/panel/key-manager.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+		const { options$, optionComparer } = this.#options;
+
+		options$
+			.pipe(
+				observeOn(asyncScheduler),
+				map((options) => options.findIndex((o) => optionComparer(o, option))),
+				filter((index) => index !== -1),
+				take(1),
+				takeUntilDestroyed(this.#destroyRef),
```
Comment (shajz) [inline packages/ng/core-select/panel/key-manager.ts:56]: Isn't `take(1)` sufficient here? This is subscribed when calling a function so I guess we want to destroy it asap?
```suggestion
				take(1),
```

**Review by shajz — APPROVED** [review]

File: packages/ng/core-select/panel/key-manager.ts (hunk 1 above)
Comment (GuillaumeNury) [reply packages/ng/core-select/panel/key-manager.ts:56]: If the observable never emits or after the component is destroyed, the Subscription would live for ever / too long

**Review by F2X — APPROVED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2964/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/20803/ui-diff)

**Pull request merged**